src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3821,6 +3820,279 @@ internal new NamedTypeSymbol CreateNativeIntegerTypeSymbol(bool signed)
+                {
+                    // bool operator ==(SomeDelegate, SomeDelegate) is legal.
+                    // bool operator !=(SomeDelegate, SomeDelegate) is legal.
+                    if (binaryKind is BinaryOperatorKind.Equal or BinaryOperatorKind.NotEqual &&
+                        csharpReturnType.SpecialType == SpecialType.System_Boolean)
+                    {
+                        return;
+                    }
+
+                    // SomeDelegate operator +(SomeDelegate, SomeDelegate) is legal.
+                    // SomeDelegate operator -(SomeDelegate, SomeDelegate) is legal.
+                    if (binaryKind is BinaryOperatorKind.Addition or BinaryOperatorKind.Subtraction &&
+                        TypeSymbol.Equals(csharpLeftType, csharpReturnType, TypeCompareKind.ConsiderEverything))
+                    {
+                        return;
+                    }
+                }
+
+                if (csharpLeftType.IsEnumType() || csharpRightType.IsEnumType())
+                {
+                    // bool operator ==(SomeEnum, SomeEnum) is legal.
+                    if (binaryKind is BinaryOperatorKind.Equal or
+                                      BinaryOperatorKind.NotEqual or
+                                      BinaryOperatorKind.GreaterThan or
+                                      BinaryOperatorKind.LessThan or
+                                      BinaryOperatorKind.GreaterThanOrEqual or
+                                      BinaryOperatorKind.LessThanOrEqual &&
+                        csharpReturnType.SpecialType is SpecialType.System_Boolean &&
+                        TypeSymbol.Equals(csharpLeftType, csharpRightType, TypeCompareKind.ConsiderEverything))
+                    {
+                        return;
+                    }
+
+                    // SomeEnum operator &(SomeEnum, SomeEnum) is legal.
+                    if (binaryKind is BinaryOperatorKind.And or
+                                      BinaryOperatorKind.Or or
+                                      BinaryOperatorKind.Xor &&
+                        TypeSymbol.Equals(csharpLeftType, csharpRightType, TypeCompareKind.ConsiderEverything) &&
+                        TypeSymbol.Equals(csharpReturnType, csharpRightType, TypeCompareKind.ConsiderEverything))
+                    {
+                        return;
+                    }
+
+                    // SomeEnum operator+(SomeEnum, int)
+                    // SomeEnum operator+(int, SomeEnum)
+                    // SomeEnum operator-(SomeEnum, int)
+                    // SomeEnum operator-(int, int)
+                    if (binaryKind is BinaryOperatorKind.Addition or BinaryOperatorKind.Subtraction)
+                    {
+                        if (csharpLeftType.IsEnumType() && csharpRightType.IsIntegralType() && TypeSymbol.Equals(csharpLeftType, csharpReturnType, TypeCompareKind.ConsiderEverything))
+                            return;
```

Comment:
>return; [](http://example.com/codeflow?start=28&length=7)

A block is required around ```if``` statement body.

Reply:
Nope :)  compiler and IDE moved to .net rules, which don't require curlies on it-statements.  IDE used to actually require this, but we stopped at request of compiler team which wanted to follow .net rules :)

Reply:
Please put them anyway. Our rule was, and I am not aware that we changed it - multiline if must use blocks

Reply:
Having blocks doesn't violate the style. Right?

Reply:
I'll add back in.  But this relaxation was made at teh request of compiler team :)   IDE always had this as a hard requirement, but we had to relax it because of this desire to match .net formatting.

Reply:
I'm just saying, we had such a rule, but it was actually removed at the request *of hte compiler team*.  Specifically just so this sort of block could be written.  It's strange to have had that happen, and now get requests to be more restrictive with the formatting, as taht was the position we were pushing for back then anyways, and which we were overruled on :)  

One of the takeaways was that when it came to formatting we would simply follow the .net rules and have tooling check/correct when in violation.  That was to help prevent these sorts of discussions from arising and it's why we settled on .net rules as well.  It feels like that has now been rolled back as we're instituting stricter rules, and we *are* spending the time in PRs discussing this, which was the thing that the team wanted to stop happening :)

Reply:
And I am just saying that we had the rule and, even though I am a member of the compiler team, I am not aware of a change made around this rule or any request that the team made about that. I am not saying that no one asked for the change on behalf of the compiler team.

Reply:
@jaredpar for the history here :)

Reply:
If you are fine with adding blocks, I am not sure what makes this thread going.

Reply:
ideally, to prevent hte discussion in future PRs.  I'm fine with adding the blocks to lubricate this PR.  Generally speaking though, i will want to continue following any legal .net style, which includes not having blocks when not required :)